test/WebJobs.Script.Tests.Integration/Diagnostics/MetricsEventManagerTests.cs

```diff
@@ -389,11 +389,8 @@ public void TimerFlush_CalledOnExpectedInterval()
                     numFlushes++;
                 });
 
-            Thread.Sleep(200);
-
-            // here we're just verifying that we're called
-            // multiple times
-            Assert.True(numFlushes >= 5, $"Expected numFlushes >= 5; Actual: {numFlushes}");
+            // here we're just verifying that we're called multiple times
+            await TestHelpers.Await(() => numFlushes >= 5, 2000, 100, userMessageCallback: () => $"Expected numFlushes >= 5; Actual: {numFlushes}");                        
```

Comment:
nit: Can you add the argument names to the timeout/polling arguments (assuming that's what they are)?